src/modules/accommodations/entities/accommodation.entity.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  AccommodationCategory,
+  AccommodationStatus,
+} from 'src/shared/constants';
+import {
+  Column,
+  CreateDateColumn,
+  Entity,
+  OneToMany,
+  PrimaryGeneratedColumn,
+  UpdateDateColumn,
+} from 'typeorm';
+import { Amenity } from './amenity.entity';
+
+@Entity()
+export class Accommodation {
+  @PrimaryGeneratedColumn('uuid')
+  id: string;
+
+  @Column({
+    type: 'varchar',
+    length: 125,
+  })
+  name: string;
+
+  @Column({
+    type: 'varchar',
+    length: 255,
+  })
+  description: string;
+
+  @Column({
+    type: 'varchar',
+    length: 200,
+  })
+  location: string;
+
+  @Column({
+    name: 'price_per_night',
+    type: 'decimal',
+  })
+  price_per_night: number;
+
+  @Column({
+    type: 'enum',
+    enum: AccommodationCategory,
+    array: true,
+    default: [AccommodationCategory.HOTEL],
```

Comment:
I think we can omit the default value here and leave it to the user to always choose from one of the available categories. If someone creates an accommodation record for their Apartment, it might be mistakenly classified as Hotel if they don't specify category.